app-service-msi-keyvault-python-v4/README.md

```diff
@@ -0,0 +1,169 @@
+or [the portal](https://azure.microsoft.com/documentation/articles/resource-group-create-service-principal-portal/)
+and grant it the same access.
+
+
+## Local dev installation
+
+1.  If you don't already have it, [install Python](https://www.python.org/downloads/).
+
+    This sample (and the SDK) is compatible with Python 2.7, 3.3, 3.4, 3.5 and 3.6.
+
+2.  We recommend that you use a [virtual environment](https://docs.python.org/3/tutorial/venv.html)
+    to run this example, but it's not required.
+    Install and initialize the virtual environment with the "venv" module on Python 3 (you must install [virtualenv](https://pypi.python.org/pypi/virtualenv) for Python 2.7):
+
+    ```
+    python -m venv mytestenv # Might be "python3" or "py -3.6" depending on your Python installation
+    cd mytestenv
+    source bin/activate      # Linux shell (Bash, ZSH, etc.) only
+    ./scripts/activate       # PowerShell only
+    ./scripts/activate.bat   # Windows CMD only
+    ```
+
+1.  Clone the repository.
+
+    ```
+    git clone https://github.com/Azure-Samples/app-service-msi-keyvault-python.git
+    ```
+
+2.  Install the dependencies using pip.
+
+    ```
+    cd app-service-msi-keyvault-python-v4
+    pip install -r requirements.txt
+    ```
+
+3.  Set up the environment variable `KEY_VAULT_URL` with your KeyVault URL of replace the variable in the example file.
+
+1. Export these environment variables into your current shell or update the credentials in the example file.
+
+    ```
+    export AZURE_TENANT_ID={your tenant id}
+    export AZURE_CLIENT_ID={your client id}
+    export AZURE_CLIENT_SECRET={your client secret}
+    ```
+
+1. Run the sample.
+
+    ```
+    python example.py
+    ```
+
+1. This sample exposes two endpoints:
+  
+   - `/ping` : This just answers "hello world" and is a good way to test if your packages are installed correctly without testing Azure itself.
+   - `/` : The MSI sample itself
+
+## Installation on Azure
+
+1. Set the `KEY_VAULT_URL` environment variable using the "Application Settings" of your WebApp.
+
+1. Connect to the [Kudu console](https://github.com/projectkudu/kudu/wiki/Kudu-console) and install the dependencies. If you installed the Python 3.6.2x86 extension, the command line will be:
+
+```shell
+D:\home\python362x86\python.exe -m pip install -r D:\home\site\wwwroot\requirements.txt
+```
+
+> For automation purpose, you might use the [Kudu RestAPI](https://github.com/projectkudu/kudu/wiki/REST-API#command)
+
+3. This repo is ready to be deployed using local git. Read this tutorial to get more information on [how to push using local git with CLI 2.0](https://docs.microsoft.com/azure/app-service/app-service-web-get-started-python#push-to-azure-from-git)
+
+## At a glance
+
+A default credential capable of handling most Azure SDK authentication scenarios.When environment variable configuration is present, it authenticates as a service principal. Then, the code is simply:
```

Comment:
This is just a couple sentences from the azure-identity docs. Should this section describe what the sample is intended to do?